File: hermes/src/main/java/org/eclipse/paho/android/service/NetworkConnectionReceiver.java
package org.eclipse.paho.android.service;

import android.annotation.SuppressLint;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.PowerManager;
import android.util.Log;

/**
 * 网络变化广播接收器
 * @author zhouL
 * @date 2023/9/19
 */
public class NetworkConnectionReceiver extends BroadcastReceiver {

    private final static String TAG = "NetworkReceiver";

    private NetworkListener mNetworkListener = null;

    @Override
    @SuppressLint("WakelockTimeout")
    public void onReceive(Context context, Intent intent) {
        String lockTag = context.getPackageName() + ":" + "MQTT";
        PowerManager pm = (PowerManager) context.getSystemService(Context.POWER_SERVICE);
        PowerManager.WakeLock wl = pm.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, lockTag);
        wl.acquire();
        boolean isOnline = MqttUtils.isOnline(context);
        Log.d(TAG, "network changed isOnline = " + isOnline);
        if (mNetworkListener != null){
            mNetworkListener.onNetworkChanged(isOnline);
        }
        wl.release();
    }

    public void setOnNetworkListener(NetworkListener listener){
        mNetworkListener = listener;
    }

    public interface NetworkListener {
        void onNetworkChanged(boolean isOnline);
    }

}

File: hermes/src/main/java/com/lodz/android/hermes/mqtt/base/sender/NetworkConnectionReceiver.kt
package com.lodz.android.hermes.mqtt.base.sender

import android.annotation.SuppressLint
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.PowerManager
import com.lodz.android.hermes.modules.HermesLog
import com.lodz.android.hermes.mqtt.base.utils.MqttUtils

/**
 * 网络变化广播接收器
 * @author zhouL
 * @date 2023/10/12
 */
class NetworkConnectionReceiver : BroadcastReceiver() {
    companion object {
        private const val TAG = "NetworkReceiver"
    }

    private var mListener: NetworkListener? = null

    @SuppressLint("WakelockTimeout")
    override fun onReceive(context: Context, intent: Intent?) {
        val lockTag = "${context.packageName}:MQTT"
        val pm = context.getSystemService(Context.POWER_SERVICE) as? PowerManager
        val wl = pm?.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, lockTag)
        wl?.acquire()
        val isOnline = MqttUtils.isOnline(context)
        HermesLog.d(TAG, "network changed isOnline = $isOnline")
        mListener?.onNetworkChanged(isOnline)
        wl?.release()
    }

    fun setOnNetworkListener(listener: NetworkListener?) {
        mListener = listener
    }

    interface NetworkListener {
        fun onNetworkChanged(isOnline: Boolean)
    }
}